pkg/app/AppListingService.go

```diff
@@ -243,6 +249,28 @@ func (impl AppListingServiceImpl) FetchOverviewAppsByEnvironment(envId, limit, o
 		if lastDeployed != nil {
 			envContainer.LastDeployedImage = lastDeployed.LastDeployedImage
 			envContainer.LastDeployedBy = lastDeployed.LastDeployedBy
+
+			artifactDetails = append(artifactDetails, lastDeployed.CiArtifactId)
+			envContainer.CiArtifactId = lastDeployed.CiArtifactId
+		}
+	}
+
+	artifactWithGitCommit, err := impl.fetchCiArtifactAndGitTriggersMap(artifactDetails)
+	if err != nil {
+		impl.Logger.Errorw("failed to fetch Artifacts to git Triggers ", "err", err, "envId", envId)
+		return resp, err
+	}
+	for _, envContainer := range envContainers {
+		if envContainer.CiArtifactId > 0 {
+			if commits, ok := artifactWithGitCommit[envContainer.CiArtifactId]; ok && commits != nil {
+				envContainer.Commits = commits
+			} else {
+				envContainer.Commits = []string{}
+
+			}
+
+		} else {
+			envContainer.Commits = []string{}
```

Comment:
This can be shortened to:-
envContainer.Commits = []string{}
if commits, ok := artifactWithGitCommit[envContainer.CiArtifactId]; ok && commits != nil {
				envContainer.Commits = commits
}